picamera2/picamera2.py

```diff
@@ -293,6 +296,30 @@ def open_camera(self) -> None:
         else:
             raise RuntimeError("Failed to initialize camera")
 
+    def read_formats(self) -> None:
+        """Read the available sensor formats
+        """
+        raw_config = self.camera.generate_configuration([libcamera.StreamRole.Raw])
+        raw_formats = raw_config.at(0).formats
+        self.available_formats = []
+
+        for pix in raw_formats.pixel_formats:
+            fmt = SensorFormat(str(pix))
+            all_format = {}
+            all_format["format"] = fmt
+            all_format["unpacked"] = fmt.unpacked
+            all_format["bit_depth"] = fmt.bit_depth
+            for size in raw_formats.sizes(pix):
+                cam_format = all_format.copy()
+                cam_format["size"] = tuple([int(i) for i in str(size).split("x")])
```

Comment:
Getting the dimensions by turning this into a string and splitting it feels slightly more awkward than I would have expected... is this a libcamera.Size object? If so it should have a width and height field.